src/js/common/schemaform/ArrayField.jsx

```diff
@@ -0,0 +1,324 @@
+
+import { errorSchemaIsValid } from './helpers';
+
+const Element = Scroll.Element;
+const scroller = Scroll.scroller;
+
+const scrollToFirstError = () => {
+  setTimeout(() => {
+    const errorEl = document.querySelector('.usa-input-error, .input-error-date');
+    if (errorEl) {
+      const position = errorEl.getBoundingClientRect().top + document.body.scrollTop;
+      Scroll.animateScroll.scrollTo(position - 10, {
+        duration: 500,
+        delay: 0,
+        smooth: true
+      });
+      focusElement(errorEl);
+    }
+  }, 100);
+};
+
+/* Non-review growable table (array) field */
+export default class ArrayField extends React.Component {
+  constructor(props) {
+    super(props);
+
+    /*
+     * We always want to show one empty item in the array, so add that if necessary
+     *
+     * It will not get saved unless a user changes data in it
+     */
+    const formData = Array.isArray(props.formData) ? props.formData : null;
+    let items;
+    if (!formData || formData.length === 0) {
+      items = [getDefaultFormState(props.schema.items, null, props.registry.definitions) || {}];
+    } else {
+      items = getDefaultFormState(props.schema, formData, props.registry.definitions) || [];
+    }
+
+    /*
+     * We're keeping the editing state in local state because it's easier to manage and
+     * doesn't need to persist from page to page
+     */
+    this.state = {
+      items,
+      editing: items.map(() => false)
+    };
+
+    this.onItemChange = this.onItemChange.bind(this);
+    this.handleAdd = this.handleAdd.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleUpdate = this.handleUpdate.bind(this);
+    this.handleRemove = this.handleRemove.bind(this);
+    this.onItemBlur = this.onItemBlur.bind(this);
+    this.scrollToTop = this.scrollToTop.bind(this);
+    this.scrollToRow = this.scrollToRow.bind(this);
+  }
+
+  /*
+   * Based on the default react-jsonschema-form optimizations, except it handles state/props syncing
+   * with formData
+   */
+  shouldComponentUpdate(nextProps, nextState) {
+    const propsWithoutDataUnchanged = deepEquals(_.omit('formData', this.props), _.omit('formData', nextProps));
+    const stateUnchanged = deepEquals(this.state, nextState);
+
+    // Because there's local state for the form data, we sometimes see form data updated in props that is already
+    // in state and we don't need to update
+    if (propsWithoutDataUnchanged && stateUnchanged && deepEquals(nextProps.formData, nextState.items)) {
+      return false;
+    }
+
+    return true;
+  }
+
+  onItemChange(indexToChange, value) {
+    // Replace the item in local state first
+    const newState = _.set('items', this.state.items.map(
+      (current, index) => {
+        return index === indexToChange
+          ? value
+          : current;
+      }
+    ), this.state);
+    // Then, update the actual form data
+    this.setState(newState, () => {
+      this.props.onChange(newState.items);
```

Comment:
Basically, yes. The `onChange` call will update the overall form data help by the overall `Form` component, which will end up triggering an update to the Redux state.